### DISCUSSION TITLE
v0.12.4 - The Islands Update

### DISCUSSION THREAD

**Discussion by ThePotatoGuy** [Monika-After-Story/MonikaModDev#8099] [Announcements & Releases]
_now with 500% more floating_

# Gameplay Changes

* Islands revamped

* o31

* Hair color retouched to be less red

* Outfit mode no longer immediately applies when opening the clothes selector with outfit mode enabled

# Topic Changes

### Removals
* lilmonix3 twitter window react

# Spritepack Changes

### Additions
* Shuchiin Academy Uniform by Briar Young (@kkrosie123)
* Blue Sweater Vest by Finale
* Beige Turtleneck Sweater by Finale

### Changes
* all hair spritepacks have changed because of the hair color change

# Fixes

* typofixes

* fixed Monika's eyes not following the cursor correctly in some circumstances

* general logging issues fixed

* fixed issue where returning home after o31 triggers the wrong greeting

* fixed hitbox size for buttons in the main menu

* fixed hotkeys breaking flow in islands

# API Changes

* EXP Previewer now supports copying/pasting sprite codes and uses the sprite codes JSON
* season progpoints are now function pluggable - see #8049
* Fixed deco APIs being partially broken - deco APIs no longer require scene change or dissolve all
* config vars have been moved to `0config.rpy`
* utils have been moved to `0utils.rpy`
* Support for parallax sprites added - see sprites section below
* Islands-related functions have changed significantly - see #7745

### Holidays
* `mas_doesBackgroundHaveHolidayDeco` - checks if a bg has support for a deco
* `mas_hasO31DeskAcs` - checks if and o31 desk ACS are present
* `mas_o31HideDeskAcs` - hides all o31 desk ACS

### Window Reacts
* `mas_getWindowTitle` - gets current title for MAS window

### Logging
* `mas_logging.init_log` - creates and initializes a python `logging.Logger` object for logging. See the python docs for how to use this object for logging.
* `MASLogFormatter` - basic log formatter for MAS logs
* `MASNewlineLogFormatter` - log formatter that supports prefixing lines with a new line (basically creating line breaks)
* `MASNewlineLogAdapter` - log adapter that supports enables `MASNewlineLogFormatter` to work
* `MASExtraPropLogAdapter` - log adapter that supports using extra props when calling logging APIs

### Sprites
* Added Dynamic ACS (#8076) - ACS that uses disps instead of images, which means the ACS can vary more than usual
* `MASDynamicAccessory` - accessory class for dynamic accessories
* `MASDynamicSpriteBase` - speciality class that should be used as a parent when creating a dynamic sprite class
* `MASFilterable` - added a few helper classes and a defined `per_interact`
* `MASImageData` - port of `ImageData` from future renpy versions. Allows loading images from binary data.
* `ParallaxDecal` - represents a single decal on a parallax sprite
* `ParallaxSprite` - a parallax sprite
* `ParallaxBackground` - composites multiple `ParallaxSprite` objects together

<hr /><em>This discussion was created from the release <a href='https://github.com/Monika-After-Story/MonikaModDev/releases/tag/v0.12.4'>v0.12.4 - The Islands Update</a>.</em>

**Comment by cfgexe** [comment]
[code]
I'm sorry, but errors were detected in your script. Please correct the
errors listed below, and try again.


File "game/zz_calendar.rpy", line 1257: indentation mismatch.

Ren'Py Version: Ren'Py 6.99.12.4.2187
[/code]
